src/main/java/Ui.java

```diff
@@ -0,0 +1,146 @@
+import java.util.Scanner;
+
+/**
+ * Takes in the user input into a Duke object and displays output to the user.
+ */
+public class Ui {
+    protected String line = "    ____________________________________________________________" + "\n";
+    protected String welcomeMessage = "     Hello! I'm Duke\n     What can I do for you?\n";
+    protected String byeMessage = "     Bye. Hope to see you again soon!\n";
+    Scanner scanner;
+
+    /**
+     * Creates an Ui object to take input user input.
+     */
+    public Ui() {
+        scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Reads the next input from the user.
+     *
+     * @return the String command input by the user.
+     */
+    public String readCommand() {
+        return this.scanner.nextLine();
+    }
+
+    /**
+     * Returns the welcome message when a Duke object is started.
+     *
+     * @return String welcome message when Duke is started.
+     */
+    public String showWelcomeMessage() {
+        return line + welcomeMessage + line;
+    }
+
+    /**
+     * Returns the list of tasks in the TaskList object argument.
+     *
+     * @param tasks TaskList object with tasks to be displayed to the user in a list.
+     * @return String list of tasks in TaskList.
+     */
+    public String showListMessage(TaskList tasks) {
+        String list = line + "     Here are the tasks in your list:" + "\n";
```

Comment:
Good job on abstracting out the UI! I appreciate the use of protected variables to store `String` objects for easy modification of the code, which I have not done much myself. You could consider extending this philosophy to indentations as well, to make it easier to modify it in the future, especially as it repeatedly occurs in the code.